pkg/api/addons.go

```diff
@@ -194,7 +202,7 @@ func (cs *ContainerService) setAddonsConfig(isUpdate bool) {
 
 	defaultContainerMonitoringAddonsConfig := KubernetesAddon{
 		Name:    ContainerMonitoringAddonName,
-		Enabled: to.BoolPtr(DefaultContainerMonitoringAddonEnabled),
+		Enabled: to.BoolPtr(DefaultContainerMonitoringAddonEnabled && !cs.Properties.IsAzureStackCloud()),
```

Comment:
Carried this additional default gate over from the "last mile" "concatenate addons string" business logic area.